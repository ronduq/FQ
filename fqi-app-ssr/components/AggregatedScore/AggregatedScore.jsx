import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import get from 'lodash.get';
import Icon from '../Icon/Icon';
import LinkButton from '../LinkButton/LinkButton';

import { generateQuerystring } from '../../utils'
import s from './AggregatedScore.scss';

const AggregatedScore = ({
  retailersLabels,
  retailerRankData,
  query,
  buttonContent,
}) => {
  const { location, retailer } = query;
  return ( 
    <div className={s.aggregatedScore}>

      <div className={s.retailerRank}>
        <p className={s.title}>{get(retailersLabels, 'retailer_rank_title')}</p>
        <p className={s.desc}>{get(retailersLabels, 'single_produce_average_title')}</p>
      </div>

      <ul className={s.retailerData}>
        <li>
          <span className={s.heading}>{retailersLabels.quality}</span>
          {/* <Icon className={s.starIcon} iconName="star" /> */}
          <span className={s.score}>{get(retailerRankData, ['quality'], retailersLabels.no_value)}%</span>
        </li>
          <li>
            <span className={s.heading}>{retailersLabels.value}</span>
            {/* <Icon className={s.dollarIcon} iconName="dollar" /> */}
            <span className={s.score}>{get(retailerRankData, ['value'], retailersLabels.no_value)}</span>
          </li>
          <li>
            <span className={s.heading}>{retailersLabels.perception}</span>
            {/* <Icon className={s.diamondIcon} iconName="diamond" /> */}
            <span className={s.score}>{get(retailerRankData, ['perception'], retailersLabels.no_value)}</span>
          </li>
      </ul>

      <div className={s.linkWrapper}>
        <LinkButton 
          label={buttonContent.view_retailer_trends}
          as={`/${location}/retailers/${retailer}/trends`} 
          href={`/retailer-trends?${generateQuerystring(query)}`}
        />
      </div>
     
    </div>
       
  );
};

AggregatedScore.propTypes = {
  retailersLabels: PropTypes.shape().isRequired,
  retailerRankData: PropTypes.shape(),
  query: PropTypes.shape(),
};

AggregatedScore.defaultProps = {
  retailerRankData: {},
}

export default AggregatedScore;
