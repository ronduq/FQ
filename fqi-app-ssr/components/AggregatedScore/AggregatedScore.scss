@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';


.aggregatedScore {
  width: 100%;
  box-sizing: border-box;
  border-top: 1px solid $colour-grey-medium;
  border-bottom: 1px solid $colour-grey-medium;
  background-color: $white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px 16px 0;
  flex-direction: column;

  @include respond-to('tablet', min-width) {
    flex-direction: row;
    padding: 16px 16px;
    border: 1px solid $colour-grey-medium;
  }

  .retailerRank {
    display: flex;
    flex-direction: column;
    width: 100%;

    @include respond-to('tablet', min-width) {
      width: auto;
    }

    .title {
      @include font-family('proxima-bold');
      @include font-size('medium-20');

      @include respond-to('mobile') {
        @include font-size('medium');
      }
    }

    .desc {
      @include font-family('proxima-regular');
      @include font-size('medium');

      @include respond-to('mobile') {
        @include font-size('xsmall');
      }
    }
  }

  .retailerData {
    display: flex;
    flex-direction: column;
    width: 100%;

    @include respond-to('tablet', min-width) {
      flex-direction: row;
      width: auto;
    }

    li {
      display: flex;
      align-items: center;
      padding: 0px;
      height: 46px;
      border-bottom: 1px solid $colour-grey-medium;

      @include respond-to('tablet', min-width) {
        border-bottom: none;
        border-right: 1px solid $colour-grey-medium;
        padding: 0 16px;
        height: 72px;
      }
    
      &:first-child {
        padding-left: 0;
      }

      &:last-child {
        padding-right: 0;
        border-right: none;
      }

      .heading {
        @include font-family('proxima-bold');
        @include font-size('small');
        margin-right: 10px;

        @include respond-to('tablet', min-width) {
          margin-right: 6px;
          @include font-size('medium');
        }
      }

      .score {
        @include font-family('proxima-bold');
        @include font-size('medium');
        margin-left: auto;

        @include respond-to('tablet', min-width) {
          @include font-size('xlarge');
          margin-left: 10px;
        }
      }
    }

    .starIcon,
    .dollarIcon,
    .diamondIcon {
      color: $colour-main;
      width: 20px;
      height: 20px;
    }
   
  }

  .linkWrapper {
    text-align: right;
    @include respond-to('tablet') {
      height: 46px;
      display: flex;
      align-items: center;
      margin-left: auto;
    }
  }
}