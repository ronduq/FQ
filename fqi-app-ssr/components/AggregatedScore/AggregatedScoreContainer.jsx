import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import AggregatedScore from './AggregatedScore';
import { getSelectedRetailerRanks, getButtonsContent, } from '../../selectors/index';

const mapStateToProps = state => {
  return {
    retailerRankData: getSelectedRetailerRanks(state),
    buttonContent: getButtonsContent(state),
  }
}

export default connect(
  mapStateToProps,
  null
)(AggregatedScore);