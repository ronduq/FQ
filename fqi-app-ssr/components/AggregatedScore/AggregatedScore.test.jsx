/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import AggregatedScore from './AggregatedScore'

describe('AggregatedScore', () => {

  const props = {
    retailersLabels:{
      single_produce_average_title: 'Single produce',
      quality: 'quality',
      value: 'value',
      perception: 'perception',
    },
    query: {
      location: 'boston',
      retailer: 'aldi'
    }
  };

  it('AggregatedScore snapshot"', () => {
    const component = renderer.create(<AggregatedScore { ...props } />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

})