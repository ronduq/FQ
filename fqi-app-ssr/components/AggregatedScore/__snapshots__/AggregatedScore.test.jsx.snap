// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregatedScore AggregatedScore snapshot" 1`] = `
<div
  className="aggregatedScore"
>
  <div
    className="retailerRank"
  >
    <p
      className="title"
    >
      Retailer Rank
    </p>
    <p
      className="desc"
    >
      Single produce
    </p>
  </div>
  <ul
    className="retailerData"
  >
    <li>
      <span
        className="heading"
      >
        quality
      </span>
      <svg
        aria-hidden="true"
        className="starIcon"
        focusable="false"
      >
        <use
          xlinkHref="#star"
        />
      </svg>
      <span
        className="score"
      >
        n/a
        %
      </span>
    </li>
    <li>
      <span
        className="heading"
      >
        value
      </span>
      <svg
        aria-hidden="true"
        className="dollarIcon"
        focusable="false"
      >
        <use
          xlinkHref="#dollar"
        />
      </svg>
      <span
        className="score"
      >
        n/a
      </span>
    </li>
    <li>
      <span
        className="heading"
      >
        perception
      </span>
      <svg
        aria-hidden="true"
        className="diamondIcon"
        focusable="false"
      >
        <use
          xlinkHref="#diamond"
        />
      </svg>
      <span
        className="score"
      >
        n/a
      </span>
    </li>
  </ul>
  <div
    className="linkWrapper"
  >
    <a
      className="linkButton"
      href="/boston/retailers/aldi/trends"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      View Retailer Trends
       
      <svg
        aria-hidden="true"
        className="arrowIcon iconSmall iconOrange arrowRight"
        focusable="false"
      >
        <use
          xlinkHref="#chevron"
        />
      </svg>
    </a>
  </div>
</div>
`;
